staging/src/k8s.io/legacy-cloud-providers/azure/clients/vmclient/azure_vmclient.go

```diff
@@ -56,7 +56,12 @@ type Client struct {
 // New creates a new VirtualMachine client with ratelimiting.
 func New(config *azclients.ClientConfig) *Client {
 	baseURI := config.ResourceManagerEndpoint
-	authorizer := autorest.NewBearerAuthorizer(config.ServicePrincipalToken)
+	var authorizer autorest.Authorizer
```

Comment:
what about write like this:
```
authorizer := autorest.NewBearerAuthorizer(config.ServicePrincipalToken)
if config.MultiTenantServicePrincipalToken != nil {
    authorizer = autorest.NewMultiTenantServicePrincipalTokenAuthorizer(config.MultiTenantServicePrincipalToken)
}
```

Reply:
a better way is move authorizer  part of azclients.ClientConfig, then all client's authorizer could be initiazed at the same location, e.g. `getAzureClientConfig()`

Reply:
Thank you guys. Updated.